### PR TITLE
Classify TApplicationException as external error

### PR DESCRIPTION
**Pull request by jessesleeping**

### CODE REVIEW

File: presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java [hunk 1]
```diff
@@ -58,13 +61,22 @@ public RetryingPrestoThriftServiceProvider(@NonRetrying PrestoThriftServiceProvi
...
                         config.getMinRetrySleepTime(),
                         config.getMaxRetrySleepTime(),
                         config.getMaxRetryDuration(),
                         config.getRetryScaleFactor());
     }
 
+    private static Exception classifyException(Exception e)
+    {
+        if (e instanceof TApplicationException) {
+            return new PrestoException(THRIFT_GENERIC_REMOTE_SERVER_ERROR, "Exception raised by a remote thrift server", e);
```
Comment (electrum) [inline presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/clientproviders/RetryingPrestoThriftServiceProvider.java]: Capitalize Thrift (or remove the word)

File: presto-thrift-connector/pom.xml [hunk 2]
```diff
@@ -116,6 +116,11 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
```
Comment (electrum) [inline presto-thrift-connector/pom.xml]: Move this so it’s not mixed with the SPI dependencies

File: presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftErrorCode.java [hunk 3]
```diff
@@ -23,7 +23,8 @@
         implements ErrorCodeSupplier
 {
     THRIFT_SERVICE_CONNECTION_ERROR(1, EXTERNAL),
-    THRIFT_SERVICE_INVALID_RESPONSE(2, EXTERNAL);
+    THRIFT_SERVICE_INVALID_RESPONSE(2, EXTERNAL),
+    THRIFT_GENERIC_REMOTE_SERVER_ERROR(3, EXTERNAL);
```
Comment (electrum) [inline presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftErrorCode.java]: How about naming this `THRIFT_SERVICE_GENERIC_ERROR` to match the others?

Comment (jessesleeping) [reply presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftErrorCode.java]: How about `THRIFT_SERVICE_GENERIC_REMOTE_ERROR`? Basically we want to emphasize that it's an error in the remote thrift server side.

Comment (electrum) [reply presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftErrorCode.java]: Sounds good

**Pull request merged**